ddtrace/contrib/flask/patch.py

```diff
@@ -307,11 +307,19 @@ def traced_start_response(status_code, headers):
                 if not s.get_tag(FLASK_ENDPOINT) and not s.get_tag(FLASK_URL_RULE):
                     s.resource = u"{} {}".format(request.method, code)
 
-                s.set_tag(http.STATUS_CODE, code)
-                if 500 <= code < 600:
-                    s.error = 1
-                elif code in config.flask.get("extra_error_codes", set()):
-                    s.error = 1
+                trace_utils.set_http_meta(s, config.flask, status_code=code, response_headers=headers)
+
+                extra_error_codes = config.flask.get("extra_error_codes")
+                if extra_error_codes:
+                    log.warning(
```

Comment:
We should probably do a `debtcollector.deprecate()` here instead. We can probably set a `removal_version` to an upcoming version.

Reply:
Done